recipe-server/client/control_new/components/recipes/Listing.js

```diff
@@ -0,0 +1,185 @@
+import LoadingOverlay from 'control_new/components/common/LoadingOverlay';
+import QueryRecipes from 'control_new/components/data/QueryRecipes';
+import ListingActionBar from 'control_new/components/recipes/ListingActionBar';
+import DataList from 'control_new/components/tables/DataList';
+import * as recipeActions from 'control_new/state/recipes/actions';
+import {
+  getRecipeListingColumns,
+  getRecipeListingCount,
+  getRecipeListingFlattenedAction,
+} from 'control_new/state/recipes/selectors';
+import {
+  getCurrentURL,
+  getQueryParameter,
+} from 'control_new/state/router/selectors';
+
+
+@connect(
+  state => ({
+    columns: getRecipeListingColumns(state),
+    count: getRecipeListingCount(state),
+    getCurrentURL: queryParams => getCurrentURL(state, queryParams),
+    ordering: getQueryParameter(state, 'ordering', '-last_updated'),
+    pageNumber: parseInt(getQueryParameter(state, 'page', 1), 10),
+    recipes: getRecipeListingFlattenedAction(state),
+    searchText: getQueryParameter(state, 'searchText'),
+    status: getQueryParameter(state, 'status'),
+  }),
+  dispatch => (
+    bindActionCreators({
+      fetchFilteredRecipesPage: recipeActions.fetchFilteredRecipesPage,
+      push: pushAction,
+    }, dispatch)
+  )
+)
+@autobind
+export default class Listing extends React.Component {
+  static propTypes = {
+    columns: PropTypes.object,
+    count: PropTypes.number,
+    fetchFilteredRecipesPage: PropTypes.func.isRequired,
+    getCurrentURL: PropTypes.func.isRequired,
+    ordering: PropTypes.string,
+    pageNumber: PropTypes.number,
+    push: PropTypes.func.isRequired,
+    recipes: PropTypes.object.isRequired,
+    searchText: PropTypes.string,
+    status: PropTypes.string,
+  };
+
+  static columnRenderers = {
+    name({ ordering }) {
+      return (
+        <Table.Column
+          title="Name"
+          dataIndex="name"
+          key="name"
+          render={Listing.renderLinkedText}
+          sortOrder={DataList.getSortOrder('name', ordering)}
+          sorter
+        />
+      );
+    },
+
+    action() {
+      return (
+        <Table.Column
+          title="Action"
+          dataIndex="action"
+          key="action"
+          render={Listing.renderLinkedText}
+        />
+      );
+    },
+
+    enabled({ status }) {
+      return (
+        <Table.Column
+          title="Enabled"
+          key="status"
+          render={(text, record) => <BooleanIcon value={record.enabled} />}
+          filters={[
+            { text: 'Enabled', value: 'enabled' },
+            { text: 'Disabled', value: 'disabled' },
+          ]}
+          filteredValue={status}
+          filterMultiple={false}
+        />
+      );
+    },
+
+    lastUpdated({ ordering }) {
+      return (
+        <Table.Column
+          title="Last Updated"
+          key="last_updated"
+          dataIndex="last_updated"
+          render={(text, record) => {
+            const lastUpdated = moment(record.last_updated);
```

Comment:
I think it's better practice to provide `moment` with the date format instead of having it guess. I'm not sure how `last_updated` is formatted though so maybe this is a non-issue.